drivers/SmartThings/zigbee-motion-sensor/src/aqara/high-precision/init.lua

```diff
@@ -0,0 +1,119 @@
+  { mfr = "LUMI", model = "lumi.motion.agl04" }
+}
+
+local CONFIGURATIONS = {
+  {
+    cluster = OccupancySensing.ID,
+    attribute = OccupancySensing.attributes.Occupancy.ID,
+    minimum_interval = 30,
+    maximum_interval = 3600,
+    data_type = OccupancySensing.attributes.Occupancy.base_type,
+    reportable_change = 1
+  },
+  {
+    cluster = PowerConfiguration.ID,
+    attribute = PowerConfiguration.attributes.BatteryVoltage.ID,
+    minimum_interval = 30,
+    maximum_interval = 3600,
+    data_type = PowerConfiguration.attributes.BatteryVoltage.base_type,
+    reportable_change = 1
+  },
+  {
+    cluster = aqara_utils.PRIVATE_CLUSTER_ID,
+    attribute = aqara_utils.FREQUENCY_ATTRIBUTE_ID,
+    minimum_interval = 30,
+    maximum_interval = 3600,
+    data_type = data_types.Uint8.ID,
+    reportable_change = 1
+  },
+  {
+    cluster = aqara_utils.PRIVATE_CLUSTER_ID,
+    attribute = aqara_utils.SENSITIVITY_ATTRIBUTE_ID,
+    minimum_interval = 30,
+    maximum_interval = 3600,
+    data_type = data_types.Uint8.ID,
+    reportable_change = 1
+  }
+}
+
+local is_aqara_products = function(opts, driver, device)
+  for _, fingerprint in ipairs(FINGERPRINTS) do
+    if device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
+      return true
+    end
+  end
+  return false
+end
+
+local function device_init(driver, device)
+  battery_defaults.build_linear_voltage_init(2.6, 3.0)(driver, device)
+
+  for _, attribute in ipairs(CONFIGURATIONS) do
+    device:add_configured_attribute(attribute)
+    device:add_monitored_attribute(attribute)
+  end
+end
+
+local function added_handler(self, device)
+  device:emit_event(capabilities.motionSensor.motion.inactive())
+  device:emit_event(capabilities.illuminanceMeasurement.illuminance(0))
+  device:emit_event(aqara_utils.detectionFrequency.detectionFrequency(aqara_utils.PREF_FREQUENCY_VALUE_DEFAULT))
+  device:emit_event(aqara_utils.sensitivityAdjustment.sensitivityAdjustment.Medium())
+  device:emit_event(capabilities.battery.battery(100))
+
+  device:send(cluster_base.write_manufacturer_specific_attribute(device,
+    aqara_utils.PRIVATE_CLUSTER_ID, aqara_utils.PRIVATE_ATTRIBUTE_ID, aqara_utils.MFG_CODE, data_types.Uint8, 1))
+  device:send(aqara_utils.read_custom_attribute(device, aqara_utils.PRIVATE_CLUSTER_ID,
+    aqara_utils.FREQUENCY_ATTRIBUTE_ID))
+  device:send(aqara_utils.read_custom_attribute(device, aqara_utils.PRIVATE_CLUSTER_ID,
+    aqara_utils.SENSITIVITY_ATTRIBUTE_ID))
+end
+
+local aqara_motion_handler = {
+  NAME = "Aqara Motion Handler",
+  lifecycle_handlers = {
+    init = device_init,
+    added = added_handler
+  },
+  capability_handlers = {
+    [aqara_utils.sensitivityAdjustmentId] = {
+      [aqara_utils.sensitivityAdjustmentCommand] = aqara_utils.sensitivity_adjustment_handler,
+    },
+    [aqara_utils.detectionFrequencyId] = {
+      [aqara_utils.detectionFrequencyCommand] = aqara_utils.detection_frequency_handler,
+    }
```

Comment:
What I meant was that this should be moved to the parent aqara driver rather than having to set it twice.

Reply:
Delete duplicate codes.